crates/indexer/src/models/collection_datas.rs

```diff
@@ -36,95 +39,90 @@ pub struct CollectionData {
+    pub description_mutable: bool,
+    pub last_transaction_version: i64,
+    // Default time columns
+    pub inserted_at: chrono::NaiveDateTime,
+}
+
 impl CollectionData {
     pub fn from_write_table_item(
         table_item: &APIWriteTableItem,
         txn_version: i64,
         table_handle_to_owner: &TableHandleToOwner,
-    ) -> anyhow::Result<Option<Self>> {
+    ) -> anyhow::Result<Option<(Self, CurrentCollectionData)>> {
         let table_item_data = table_item.data.as_ref().unwrap();
-        if table_item_data.value_type != "0x3::token::CollectionData" {
-            return Ok(None);
-        }
 
-        let value = &table_item_data.value;
-        let table_handle = table_item.handle.to_string();
-        let creator_address = table_handle_to_owner
-                .get(&TableMetadataForToken::standardize_handle(&table_handle))
-                .map(|table_metadata| table_metadata.owner_address.clone())
-                .context(format!(
-                    "version {} failed! collection creator resource was missing, table handle {} not in map {:?}",
-                    txn_version, TableMetadataForToken::standardize_handle(&table_handle), table_handle_to_owner,
-                ))?;
-        let collection_name = value["name"]
-            .as_str()
-            .map(|s| s.to_string())
-            .context(format!(
-                "version {} failed! name missing from collection {:?}",
-                txn_version, value
-            ))?;
-        let collection_name_hash = hash_str(&collection_name);
+        let maybe_collection_data = match TokenWriteSet::from_table_item_type(
+            table_item_data.value_type.as_str(),
+            &table_item_data.value,
+            txn_version,
+        )? {
+            Some(TokenWriteSet::CollectionData(inner)) => Some(inner),
+            _ => None,
+        };
+        if let Some(collection_data) = maybe_collection_data {
+            let table_handle = table_item.handle.to_string();
+            let creator_address = table_handle_to_owner
+                            .get(&TableMetadataForToken::standardize_handle(&table_handle))
+                            .map(|table_metadata| table_metadata.owner_address.clone())
+                            .context(format!(
+                                "version {} failed! collection creator resource was missing, table handle {} not in map {:?}",
+                                txn_version, TableMetadataForToken::standardize_handle(&table_handle), table_handle_to_owner,
+                            ))?;
+            let collection_data_id = CollectionDataIdType {
+                creator: creator_address,
+                name: collection_data.name,
+            };
+            let collection_data_id_hash = hash_str(&collection_data_id.to_string());
+            let collection_name = truncate_str(&collection_data_id.name, 128).to_string();
+            let metadata_uri = truncate_str(&collection_data.uri, 128).to_string();
```

Comment:
128 characters.  Isn't that a little short for a URI?

Reply:
Oh it should be 512. Good catch!